R/importM.COSMIC.R

```diff
@@ -0,0 +1,348 @@
+#'   by default.
+#' @param writeLog Controls whether writing the result to the global logfile is
+#'   enabled. TRUE by default.
+#' @return Return rCNA object of COSMIC CNV data as RDS.
+#'
+#' @family importM.COSMIC, importM.TCGA, importM.GISTIC2, importM.ProjectGenie
+#'
+#'   ## @seealso \code{\link{fastMap}} fastMap() is used internally to
+#'   map COSMIC IDs to gene symbols.
+#'
+#' @examples
+#' \dontrun{
+#' fNameCNV <- "./../test/COSMIC/testCosmicCNA.tsv"
+#'
+#' importM.COSMIC(fNameCNV,
+#'                writeToFile = FALSE,
+#'                silent = TRUE,
+#'                writeLog = FALSE)
+#' }
+#' @export
+importM.COSMIC <- function(fNameCNV,
+                           #fNameHGNC,
+                           outFName,
+                           writeToFile = FALSE,
+                           silent = FALSE,
+                           writeLog = TRUE) {
+
+    # fNameCNV <- "testCosmicCNA.tsv"
+    # outFName <- "cosmicCNV.rds"
+    # fNameHGNC <- "CosmicHGNC.tsv"
+
+    # ==== PARAMETER TYPE VALIDATION ==========================================
+
+    # Check that fNameCNV is a string.
+    if(!is.character(fNameCNV)) {
+        stop("fNameCNV is not a valid string.")
+    }
+
+    # # Check that fNameHGNC is a string.
+    # if(!is.character(fNameHGNC)) {
+    #     stop("fNameHGNC is not a valid string.")
+    # }
+
+    # Check that writeToFile is a logical.
+    if (!is.logical(writeToFile)) {
+        stop("writeToFile is not a valid logical.")
+    }
+
+    if (writeToFile) {
+        # Make sure output file name is provided if writeToFile is TRUE.
+        if (missing(outFName)) {
+            stop("No output file specified in outFName.")
+        }
+
+        # Check that outFName is a string.
+        if (!is.character(outFName)) {
+            stop("outFName is not a valid string.")
+        }
+
+        # Check that outFName is of type .rds
+        if (substr(outFName, nchar(outFName) - 3, nchar(outFName)) != ".rds") {
+            stop("outFName is not of type .rds.")
+        }
+    } else {
+        if (!missing(outFName)) {
+            stop("outFName provided, but writeToFile is FALSE.")
+        }
+    }
+
+    # Check that silent is a logical.
+    if (!is.logical(silent)) {
+        stop("silent is not a valid logicial.")
+    }
+
+    # Check that writeLog is a logical.
+    if (!is.logical(writeLog)) {
+        stop("writeLog is not a valid logicial.")
+    }
+
+
+    # ==== PARAMETER VALUE VALIDATION =========================================
+
+    # Check that file provided at path fNameCNV exists
+    if (!file.exists(fNameCNV)) {
+        stop("Parameter error:\n   File specified at fNameCNV not found.")
+    }
+
+    # # Check that file provided at path fNameHGNC exists
+    # if (!file.exists(fNameHGNC)) {
+    #     stop("Parameter error:\n   File specified at fNameHGNC not found.")
+    # }
+
+
+    ## REPLACED BY CALLS TO fastMap() ##
```

Comment:
Commented out code from development needs to be removed unless it is here for a reason. If so, that reason needs to be explained.